pkgs/development/compilers/ghc/8.4.3.nix

```diff
@@ -46,8 +48,7 @@ let
     include mk/flavours/\$(BuildFlavour).mk
     endif
     DYNAMIC_GHC_PROGRAMS = ${if enableShared then "YES" else "NO"}
-  '' + stdenv.lib.optionalString enableIntegerSimple ''
-    INTEGER_LIBRARY = integer-simple
+    INTEGER_LIBRARY = ${if enableIntegerSimple then "integer-simple" else "integer-gmp"}
```

Comment:
This is definitely a good change! This really confused me with cross-built always pulling in GMP but not ever using it (on android at least).

Reply:
(might be worth backporting too)

Reply:
Yes agreed heartily with both. Great change and please back-port!